pkg/image/api/helper.go

```diff
@@ -183,9 +183,28 @@ func ParseDockerImageReference(spec string) (DockerImageReference, error) {
 		ref.Tag = tag
 		ref.ID = id
 		break
+	case len(repoParts) > 3:
```

Comment:
A spec with these segments doesn't map to a namespace and imagestream name… it's only allowed when referring to external docker registries, not the openshift registry